librz/core/cmd/cmd_debug.c

```diff
@@ -4410,23 +4410,67 @@ RZ_IPI RzCmdStatus rz_cmd_debug_add_bp_handler(RzCore *core, int argc, const cha
 
 // dbl
 RZ_IPI RzCmdStatus rz_cmd_debug_list_bp_handler(RzCore *core, int argc, const char **argv, RzCmdStateOutput *state) {
-	RzOutputMode mode = state->mode;
-	switch (mode) {
-	case RZ_OUTPUT_MODE_STANDARD:
-		rz_bp_list(core->dbg->bp, 0);
-		break;
-	case RZ_OUTPUT_MODE_RIZIN:
-		rz_bp_list(core->dbg->bp, 1);
-		break;
-	case RZ_OUTPUT_MODE_JSON:
-		rz_bp_list(core->dbg->bp, 'j');
-		break;
-	case RZ_OUTPUT_MODE_QUIET:
-		rz_bp_list(core->dbg->bp, -1);
-		break;
-	default:
-		rz_warn_if_reached();
-		break;
+	RzBreakpointItem *b;
+	RzListIter *iter;
+	RzOutputMode mode = RZ_OUTPUT_MODE_STANDARD;
+	PJ *pj = NULL;
+	if (state) {
+		mode = state->mode;
+		pj = state->d.pj;
+		rz_cmd_state_output_array_start(state);
+	}
+	rz_list_foreach (core->dbg->bp->bps, iter, b) {
+		switch (mode) {
+		case RZ_OUTPUT_MODE_STANDARD: {
+			rz_cons_printf("0x%08" PFMT64x " - 0x%08" PFMT64x
+						 " %d %c%c%c %s %s %s %s cmd=\"%s\" cond=\"%s\" "
+						 "name=\"%s\" module=\"%s\"\n",
+				b->addr, b->addr + b->size, b->size,
+				((b->perm & RZ_BP_PROT_READ) | (b->perm & RZ_BP_PROT_ACCESS)) ? 'r' : '-',
+				((b->perm & RZ_BP_PROT_WRITE) | (b->perm & RZ_BP_PROT_ACCESS)) ? 'w' : '-',
+				(b->perm & RZ_BP_PROT_EXEC) ? 'x' : '-',
```

Comment:
i would suggest to create a static function where you give the flags and a char buffer and you set the flags accordingly.
the char buffer can be declared next to the PJ declaration as `char perms[10];`

Reply:
I have used `rz_str_rwx` instead.